virttest/utils_time.py

```diff
@@ -0,0 +1,205 @@
+                                            host_tz.get('city_timezone'))
+            else:
+                logging.warn("Can not get guest's timezone.")
+                return False
+
+        # 'check_tz_cmd' got a name of timezone that only used in Windows os,
+        # we need to check the whole list of timezone in Windows to get the
+        # corresponding UTC code.
+        elif os_type == "windows":
+            check_tz_cmd = "tzutil /g"
+            set_tz_cmd = 'tzutil /s "%s"'
+            list_tz_cmd = "tzutil /l"
+            guest_tz = []
+            guest_tzs = []
+
+            o_guest = session.cmd_output(check_tz_cmd)
+            tz_list = session.cmd_output(list_tz_cmd)
+            tz_list = tz_list.splitlines()
+            for line in tz_list:
+                # empty line
+                if not line:
+                    continue
+                # try to get the UTC code from the line
+                og = re.search(r'(?:\(UTC([+|-]\d{2}:\d{2})?)', line,
+                               re.M | re.I)
+                if og and og.group(1):
+                    og_ = re.sub(r'(\d{2}):(\d{2})', r'\1\2', og.group(1))
+                    guest_tz.append(og_)
+                    continue
+                # when it is UTC time, add timezone code manually.
+                elif og:
+                    guest_tz.append("+0000")
+                    continue
+                og_value = re.search(r'(\S+(?:\s\S+)*)', line, re.M | re.I)
+                if og_value:
+                    guest_tz.append(og_value.group(0))
+                else:
+                    logging.warn("Can not get the name of timezone correctly.")
+
+                if o_guest == guest_tz[1] and (guest_tz[0] ==
+                                               host_tz.get('code_timezone')):
+                    return True
+                guest_tzs.append(guest_tz)
+                guest_tz = []
+
+            timezone_setname = ""
+            for value in guest_tzs:
+                if host_tz.get("code_timezone") == value[0]:
+                    timezone_setname = value[1]
+                    break
+
+            status = session.cmd_status(set_tz_cmd % timezone_setname)
+    finally:
+        session.close()
+
+    if not status:
+        # boot again
+        vm.destroy()
+        vm.create(params=params)
+        vm.verify_alive()
+        return True
+    else:
+        logging.warn("Fail to update timezone in guest")
+        return False
+
+
+def set_clocksource(expected, session=None):
+    """
+    Handle host/guest's clocksource info.
+
+    :param expected: expected clocksource.
+    :param session: VM session.
+
+    :return: True if clocksource set succeed.
+    """
+    clocksource_dir = "/sys/devices/system/clocksource/clocksource0/"
+    clocksource_dir += "current_clocksource"
+    status = -1
+    try:
+        if session:
+            status = session.cmd_status(
+                    "echo %s > %s" % (expected, clocksource_dir))
```

Comment:
form the string once and reuse it `cmd = "echo %s > %s" % (expected, clocksource_dir)` here and elsewhere. same in other methods.